aten/src/ATen/core/jit_type.h

```diff
@@ -66,6 +66,13 @@ CAFFE2_API const char* typeKindToString(TypeKind kind);
 
 struct Type;
 using TypePtr = std::shared_ptr<Type>;
+using ConstTypePtr = std::shared_ptr<const Type>;
```

Comment:
Is the const really necessary? If it lands you are going to see it start to proliferate and then there are will be a bunch of people complaining they have a TypePtr but need a ConstTypePtr.